File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
MAX = 10000
result = [1] * (MAX+1)

for num in range(MAX):
    str_num = str(num)
    temp = 0
    for _ in range(len(str_num)):
        temp += int(str_num[_])
    temp+= num
    if temp <= MAX:
        result[temp]=0

while True:
    try:
        min_index = result.index(1)
        print(min_index)
        result[min_index]=2
    except:
        break

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.java
public class Main {
    static int MAX = 10000;

    public static int selfNum(int num) {
        int answer = num;

        while(num>0){
            answer += num%10;
            num= num/10;
        }

        return answer;
    }

    public static void main(String[] args) {
        int[] answer = new int[MAX + 1];
        for (int index = 1; index <= MAX; index++) {
            int result = selfNum(index);
            if (result <= MAX) {
                answer[result] = 1;
            }
        }

        StringBuilder sb = new StringBuilder();

        for (int index = 1; index <= MAX; index++) {
            if (answer[index] != 1) {
                sb.append(index).append('\n');
            }
        }
        System.out.println(sb);
    }
}